File: app/src/main/java/com/masterjefferson/flockulator/domain/UseCase.java
package com.masterjefferson.flockulator.domain;

/**
 * ${FILE_NAME}
 * Created by jeff on 9/21/17.
 */

public abstract class UseCase<A, R> {
  public abstract R run(A args);

  public R run() {
    return run(null);
  }
}

File: app/src/main/java/com/masterjefferson/flockulator/domain/UseCase.kt
package com.masterjefferson.flockulator.domain

/**
 * ${FILE_NAME}
 * Created by jeff on 9/21/17.
 */

abstract class UseCase<A, R> {
    abstract fun run(args: A?): R?

    fun run(): R? {
        return run(null)
    }
}
